mon-pix/tests/unit/routes/account-recovery/update-sco-record-test.js

```diff
@@ -150,11 +150,85 @@ describe('Unit | Route | account-recovery | update sco record', function() {
           // then
           return promise.then((result) => {
             expect(result.errorMessage).to.equal(this.intl.t('api-error-messages.internal-server-error'));
-            expect(result.showReturnToHomeButton).to.be.true;
+            expect(result.showBackToHomeButton).to.be.true;
           });
         });
       });
+    });
+  });
+
+  describe('#authenticateUser', function() {
+
+    it('should call session authenticate method with password and email', async function() {
+      // given
+      const email = 'user@example.net';
+      const password = 'Password123';
+      const scope = 'mon-pix';
+
+      const sessionStub = Service.create({
+        authenticate: sinon.stub(),
+      });
+
+      const route = this.owner.lookup('route:account-recovery/update-sco-record');
+      route.session = sessionStub;
+
+      // when
+      await route.authenticateUser({ email, password });
+
+      // then
+      sinon.assert.calledWith(route.session.authenticate, 'authenticator:oauth2', {
+        login: email,
+        password,
+        scope,
+      });
+    });
+
+    context('when user is already authenticated', function() {
 
+      it('should load currentUser and replace by index route', async function() {
+        // given
+        const email = 'user@example.net';
+        const password = 'Password123';
+
+        const sessionStub = Service.create({
+          authenticate: sinon.stub().resolves(),
+          invalidate: sinon.stub().resolves(),
+          isAuthenticated: true,
+        });
+
+        const route = this.owner.lookup('route:account-recovery/update-sco-record');
+        route.session = sessionStub;
+
+        // when
+        await route.authenticateUser({ email, password });
+
+        // then
+        sinon.assert.called(route.session.invalidate);
```

Comment:
**question**
Même question que plus haut, je n'arrive pas à comprendre le lien entre la description du `it` et le `then` : 
invalider une session c'est en fait charger l'utilisateur et le rediriger ici ? 
Peut être que le terme `invalidate` est flou pour moi ou alors il n'est pas assez explicite ?

Reply:
Done